yang/frr-pim.yang

```diff
@@ -0,0 +1,446 @@
+module frr-pim {
+  yang-version "1.1";
+  namespace "http://frrouting.org/yang/pim";
+
+  prefix frr-pim;
+
+  import frr-interface {
+    prefix frr-interface;
+  }
+
+  import frr-routing {
+    prefix "frr-rt";
+  }
+
+  import ietf-routing-types {
+    prefix "rt-types";
+  }
+
+  import ietf-inet-types {
+    prefix "inet";
+  }
+
+  organization
+    "Free Range Routing";
+
+  contact
+    "FRR Users List:       <mailto:frog@lists.frrouting.org>
+     FRR Development List: <mailto:dev@lists.frrouting.org>";
+
+  description
+    "The module defines a collection of YANG definitions common for
+     PIM (Protocol Independent Multicast) model.";
+
+  revision 2017-03-09 {
+    description
+      "Initial revision.";
+    reference
+      "RFC XXXX: A YANG Data Model for PIM";
+  }
+
+  identity pimd {
+    base frr-rt:routing-protocol;
+    description
+      "'Pim' routing pseudo-protocol.";
+  }
+
+  typedef plist-ref {
+    type string;
+  }
+
+  /*
+   * Groupings
+   */
+
+  grouping global-pim-config-attributes {
+    description
+      "A grouping defining pim global attributes.";
+
+    leaf ecmp {
+      type empty;
+      description
+        "Enable PIM ECMP.";
+    }
+
+    leaf ecmp-rebalance {
+      type empty;
+      description
+        "Enable PIM ECMP Rebalance.";
+    }
+
+    leaf join-prune-interval {
+      type uint16 {
+        range "60..600";
+      }
+      default "60";
+      description
+        "Join Prune Send Interval in seconds.";
+    }
+
+    leaf keep-alive-timer {
+      type uint16 {
+        range "31..60000";
+      }
+      default "210";
+      description
+        "Keep alive Timer in seconds.";
+    }
+
+    leaf rp-keep-alive-timer {
+      type uint16 {
+        range "31..60000";
+      }
+      default "210";
+      description
+        "RP keep alive Timer in seconds.";
+    }
+
+    leaf packets {
+      type uint8 {
+        range "1..100";
+      }
+      default "3";
+      description
+        "Number of packets.";
```

Comment:
This description is basically meaningless. I'll still merge this but we need to fix it.

Reply:
addressed